awx/main/models/inventory.py

```diff
@@ -1024,7 +1024,9 @@ def cloud_credential_validation(source, cred):
             # If a credential was provided, it's important that it matches
             # the actual inventory source being used (Amazon requires Amazon
             # credentials; Rackspace requires Rackspace credentials; etc...)
-            if source.replace('ec2', 'aws') != cred.kind:
+            if source == 'vmware_esxi' and source.replace('vmware_esxi', 'vmware') != cred.kind:
```

Comment:
@thedoubl3j original commit into tower stable-2.6 here https://github.com/ansible/tower/commit/d35d7f62ec461964b7f4711e81b51904ff05e550

does this make sense in awx?

Reply:
As a `PluginFileInjector`, presumably it would go in

https://github.com/ansible/awx-plugins/blob/devel/src/awx_plugins/inventory/plugins.py

and it is not now.

Reply:
these should not be here. context wise, the work was _only_ done in controller. this work has not yet been ported to awx-plugins. we will now have to re-implement these into awx-plugins as part of feature parody.

Reply:
We have a plan, new issue to track, and put TODOs in code.